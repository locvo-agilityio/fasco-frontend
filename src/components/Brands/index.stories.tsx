import type { Meta, StoryObj } from '@storybook/react';

// Components
import Brands from '.';

const meta = {
  title: 'Components/Brands',
  component: Brands,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Brands>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
