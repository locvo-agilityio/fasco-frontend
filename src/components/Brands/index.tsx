import Image from 'next/image';
import { useMemo } from 'react';

// Constants
import { BRANDS } from '@/constants';

const Brands = () => {
  const renderBrands = useMemo(
    () =>
      BRANDS.map((brand) => (
        <div key={brand.name} className="w-44 h-7">
          <div className="relative w-full h-full">
            <Image
              fill
              src={brand.image.src}
              alt={brand.name}
              key={brand.name}
              sizes="50vw"
              priority
              style={{ objectFit: 'contain' }}
            />
          </div>
        </div>
      )),
    [],
  );

  return (
    <div className="flex w-full justify-between py-20">{renderBrands}</div>
  );
};
export default Brands;
