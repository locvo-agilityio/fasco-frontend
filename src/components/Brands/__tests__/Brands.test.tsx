import { render } from '@testing-library/react';

// Components
import Brands from '..';

describe('Brands', () => {
  it('should render correctly', () => {
    const { container } = render(<Brands />);

    expect(container).toMatchSnapshot();
  });
});
