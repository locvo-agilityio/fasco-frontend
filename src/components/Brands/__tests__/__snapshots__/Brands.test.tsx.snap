// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands should render correctly 1`] = `
<div>
  <div
    class="flex w-full justify-between py-20"
  >
    <div
      class="w-44 h-7"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Chanel"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-44 h-7"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Louis Vuitton"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-44 h-7"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="prada"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-44 h-7"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Calvin Klein"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
    <div
      class="w-44 h-7"
    >
      <div
        class="relative w-full h-full"
      >
        <img
          alt="Denim"
          data-nimg="fill"
          decoding="async"
          sizes="50vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
        />
      </div>
    </div>
  </div>
</div>
`;
